TransferMethod/Sources/UpdateTransferMethodPresenter.swift

```diff
@@ -0,0 +1,313 @@
+// including without limitation the rights to use, copy, modify, merge, publish, distribute,
+// sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or
+// substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import HyperwalletSDK
+#if !COCOAPODS
+import Common
+import TransferMethodRepository
+#endif
+
+protocol UpdateTransferMethodView: class {
+    func fieldValues() -> [(name: String, value: String)]
+    func dismissProcessing(handler: @escaping () -> Void)
+    func hideLoading()
+    func areAllFieldsValid() -> Bool
+    func notifyTransferMethodUpdated(_ transferMethod: HyperwalletTransferMethod)
+    func showConfirmation(handler: @escaping () -> Void)
+    func showError( title: String, message: String)
+    func showError(_ error: HyperwalletErrorType,
+                   pageName: String,
+                   pageGroup: String,
+                   _ retry: (() -> Void)?)
+    func showLoading()
+    func showProcessing()
+    func reloadData(_ fieldGroups: [HyperwalletFieldGroup])
+    func showFooterViewWithUpdatedSectionData(for sections: [UpdateTransferMethodSectionData])
+}
+
+final class UpdateTransferMethodPresenter {
+    private weak var view: UpdateTransferMethodView?
+    private let errorTypeApi = "API"
+    private let createdConfirmationPageName = "transfer-method:update:transfer-method-updated"
+    private let pageLink = "update-transfer-method"
+    private let transferMethodUpdatedGoal = "transfer-method-updated"
+    static let updateTransferMethodPageGroup = "update-transfer-method"
+    static let updateTransferMethodPageName = "transfer-method:update:collect-transfer-method-information"
+    private let hyperwalletInsights: HyperwalletInsightsProtocol
+    let transferMethod: HyperwalletTransferMethod?
+    var sectionData = [UpdateTransferMethodSectionData]()
+
+    private lazy var transferMethodUpdateConfigurationRepository = {
+        TransferMethodRepositoryFactory.shared.transferMethodUpdateConfigurationRepository()
+    }()
+
+    private lazy var transferMethodRepository = {
+        TransferMethodRepositoryFactory.shared.transferMethodRepository()
+    }()
+
+    init(_ view: UpdateTransferMethodView,
+         _ transferMethod: HyperwalletTransferMethod,
```

Comment:
this will be just transferMethodToken: String

Reply:
Updated